src/openrct2/paint/sprite/Paint.Sprite.cpp

```diff
@@ -49,22 +49,22 @@ void sprite_paint_setup(paint_session* session, const uint16_t x, const uint16_t
 
     const bool highlightPathIssues = (session->ViewFlags & VIEWPORT_FLAG_HIGHLIGHT_PATH_ISSUES);
 
-    for (const rct_sprite* spr = get_sprite(sprite_idx); sprite_idx != SPRITE_INDEX_NULL;
-         sprite_idx = spr->generic.next_in_quadrant)
+    for (; sprite_idx != SPRITE_INDEX_NULL;)
```

Comment:
Well, this is basically a `while (sprite_idx != SPRITE_INDEX_NULL)` now, isn't it?

Reply:
I was trying to get all the quadrant loops to the same style and then replace them with some sort of ranged for loop. But yes this one is probably special.